### ISSUE TITLE
Chip delete button tap target does not meet platform recommended tap target size

### ISSUE THREAD

**Issue by NickAb** [flutter/flutter#94098]
Reopening #42455 `Chip delete button tap target is too small.` 

## Steps to Reproduce

1. Add `androidTapTargetGuideline` test case to `packages/flutter/test/material/chip_test.dart`. See code below.
    ```dart
      testWidgets('delete button tap target complies with minimum tap target guideline', (WidgetTester tester) async {
        final SemanticsHandle handle = tester.ensureSemantics();
        final UniqueKey deleteKey = UniqueKey();
        await tester.pumpWidget(
          _wrapForChip(
            child: Column(
              children: <Widget>[
                Chip(
                  label: const Text('Label'),
                  deleteIcon: Icon(Icons.delete, key: deleteKey),
                  onDeleted: () { },
                ),
              ],
            ),
          ),
        );

        await expectLater(tester, meetsGuideline(androidTapTargetGuideline));

        handle.dispose();
      });
    ```
2.  Execute `cd packages/flutter` and `flutter test flutter/test/material/chip_test.dart` 

**Expected results:** 
- All tests pass

**Actual results:** 
- `delete button tap target complies with minimum tap target guideline [E]` fails with 
    ```
    Expected: Tappable objects should be at least Size(48.0, 48.0)
    Actual: <Instance of 'WidgetTester'>
    Which: SemanticsNode#5(Rect.fromLTRB(90.0, 12.0, 114.0, 36.0), actions: [tap], flags: [isButton,
  isFocusable], label: "Delete", textDirection: ltr): expected tap target size of at least Size(48.0,
  48.0), but found Size(24.0, 24.0)
            See also: https://support.google.com/accessibility/android/answer/7101858?hl=en
    ```

<details><summary>Flutter Doctor</summary>

```
[✓] Flutter (Channel unknown, 2.6.0-12.0.pre.805, on Debian GNU/Linux rodete 5.10.46-5rodete1-amd64, locale en_US.utf8)
    • Flutter version 2.6.0-12.0.pre.805 at /usr/local/google/home/abalov/flutter
    • Upstream repository unknown
    • Framework revision aee9ade8e3 (60 minutes ago), 2021-11-23 04:03:03 -0500
    • Engine revision f4cb61cefc
    • Dart version 2.16.0 (build 2.16.0-36.0.dev)
    • DevTools version 2.8.0

```

</details>

**Comment by NickAb** [comment]
Any updates?

**Comment by TahaTesser** [comment]
I was looking to fix this today but looks like the tappable area is matching native. It is equally small on native Android 

![photo_2022-04-04 11 33 41](https://user-images.githubusercontent.com/48603081/161512184-896a9539-ca75-41b1-817e-e8681b5c6af7.jpeg)

**Comment by benjaminwg2017** [comment]
This impacts our app as well, see internal b/411597987

**Comment by chunhtai** [comment]
according https://m3.material.io/components/chips/specs

The close button can have minimum 48 tap target

**Comment by Piinks** [comment]
It looks like this depends on the device you are using based on the default visual density that is applied to the components in your app.
I found that when using MaterialTapTargetSize.padded + VisualDensity.standard, the hit area of the delete button is inflated to honor the minimum tap target size as expected. That is a workaround that can be used for now. These properties can be configured on a given chip, or set in the ThemeData to apply to the whole app.

We are looking into what the impact would be to change the default to this so it can work out of the box as expected. We noticed this is not specific to this widget (https://github.com/flutter/flutter/issues/123528) and should be investigated fro a systemic change to fix for all the affected components.

```dart
import 'package:flutter/material.dart';

void main() => runApp(const ChipApp());

class ChipApp extends StatelessWidget {
  const ChipApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(
        colorSchemeSeed: const Color(0xff6750a4),
        useMaterial3: true,
        // workaround - visualDensity also available here for app-wide config
        // workaround - materialTapTargetSize also available here for app-wide config
      ),
      home: const InputChipExample(),
    );
  }
}

class InputChipExample extends StatefulWidget {
  const InputChipExample({super.key});

  @override
  State<InputChipExample> createState() => _InputChipExampleState();
}

class _InputChipExampleState extends State<InputChipExample> {
  int inputs = 3;
  int? selectedIndex;


  @override
  Widget build(BuildContext context) {// { .pressed }
    return Scaffold(
      appBar: AppBar(title: const Text('InputChip Sample')),
      body: Center(
        child: Column(
          mainAxisSize: MainAxisSize.min,
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Wrap(
              alignment: WrapAlignment.center,
              spacing: 5.0,
              children:
                  List<Widget>.generate(inputs, (int index) {
                    return InputChip(
                      visualDensity: VisualDensity.standard, // Workaround - also in ThemeData
                      materialTapTargetSize: MaterialTapTargetSize.padded, // Workaround - also in ThemeData
                      label: Text('Person ${index + 1}'),
                      selected: selectedIndex == index,
                      onSelected: (bool selected) {
                        setState(() {
                          if (selectedIndex == index) {
                            selectedIndex = null;
                          } else {
                            selectedIndex = index;
                          }
                        });
                      },
                      onDeleted: () {
                        setState(() {
                          inputs = inputs - 1;
                        });
                      },
                    );
                  }).toList(),
            ),
            const SizedBox(height: 10),
            ElevatedButton(
              onPressed: () {
                setState(() {
                  inputs = 3;
                });
              },
              child: const Text('Reset'),
            ),
          ],
        ),
      ),
    );
  }
}
```

**Comment by Piinks** [comment]
Marked P2 since we were able to provide a solution to unblock folks for now.

**Comment by bleroux** [comment]
@Piinks Interesting. For the moment, I'm not sure this issue and https://github.com/flutter/flutter/issues/123528 are directly related.

The test from this issues's [description](https://github.com/flutter/flutter/issues/94098#issue-1061050719) does not specify a target platform so it runs on Android (and I can confirm the proposed test is still failing on master). 
On Android visual density defaults to `VisualDensity.standard` and `ThemeData.materialTapTargetSize` defaults to `MaterialTapTargetSize.padded`.

See https://github.com/flutter/flutter/blob/48f87a5fe76aa65f89f37cc716e50b34933e78e9/packages/flutter/lib/src/material/theme_data.dart#L388-L397

So setting those values does not change the result of the given test.

What I noticed when running your code sample from https://github.com/flutter/flutter/issues/94098#issuecomment-2843400016 on Android (with or without the workaround)
- the hit area of the button is 48x48
- when activating TalkBack the semantics rect is way smaller and hard to hit.

So it seems be a problem related to the Semantic object surrounding the delete icon.
I will study Chip implementation and share my findings. First thing I noticed is that Chip does not rely on a Material button for the delete button:

https://github.com/flutter/flutter/blob/48f87a5fe76aa65f89f37cc716e50b34933e78e9/packages/flutter/lib/src/material/chip.dart#L1267-L1315

**Comment by bleroux** [comment]
chip.dart implements a custom hit testing strategy which explains why tapping outside of the Icon works.

The code below also contains interesting comments to explain the reasonning:

https://github.com/flutter/flutter/blob/48f87a5fe76aa65f89f37cc716e50b34933e78e9/packages/flutter/lib/src/material/chip.dart#L2389-L2426


In conclusion, there is some logic to expand the hit area of the delete button, but it won't change the Semantics rect (which is the icon rect).
I will experiment a possible solution for this.
